### PR TITLE
Remove EventListenerRecyclingInterval from EventCountersCollectorOptions

### LINKED ISSUE #4177: Remove EventCountersCollectorOptions.EventListenerRecyclingInterval
We don't need `EventCountersCollectorOptions.EventListenerRecyclingInterval` since it's intended to be used on .NET 5 only.

### PR DESCRIPTION
**Pull request by xakep139**
This PR fixes #4177 

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/extensions/pull/4178)

### CODE REVIEW

File: src/Libraries/Microsoft.Extensions.Telemetry/Microsoft.Extensions.Telemetry.json [hunk 1]
```diff
@@ -15,10 +15,6 @@
           "Member": "System.Collections.Generic.IDictionary<string, System.Collections.Generic.ISet<string>> Microsoft.Extensions.Telemetry.Metering.EventCountersCollectorOptions.Counters { get; set; }",
           "Stage": "Stable"
         },
-        {
-          "Member": "System.TimeSpan Microsoft.Extensions.Telemetry.Metering.EventCountersCollectorOptions.EventListenerRecyclingInterval { get; set; }",
-          "Stage": "Stable"
```
Comment (xakep139) [inline src/Libraries/Microsoft.Extensions.Telemetry/Microsoft.Extensions.Telemetry.json:20]: @geeknoid hopefully it's not too late to do such things

**Comment by RussKie** [comment]
This change is a breaking changes, which is typically not allowed. We've already made some API breaking changes in this release, so I _suppose_ this is still acceptable. However,  @joperezr  we need to create clarity on this.

@xakep139 to fix the build the removed API have to be recorded in PublicApi.Unshipped.txt as per https://github.com/dotnet/extensions/tree/main/eng/ApiCompatibility.

File: src/Libraries/Microsoft.Extensions.Telemetry/Microsoft.Extensions.Telemetry.json (hunk 1 above)
Comment (geeknoid) [reply src/Libraries/Microsoft.Extensions.Telemetry/Microsoft.Extensions.Telemetry.json:20]: Nope, not too late

**Comment by geeknoid** [comment]
> This change is a breaking changes, which is typically not allowed. We've already made some API breaking changes in this release, so I _suppose_ this is still acceptable. However, @joperezr we need to create clarity on this.
> 
> @xakep139 to fix the build the removed API have to be recorded in PublicApi.Unshipped.txt as per https://github.com/dotnet/extensions/tree/main/eng/ApiCompatibility.

This repo will necessarily continue to have many breaking changes until November.

**Review by geeknoid — APPROVED** [review]

**Pull request merged**